### PR TITLE
feat(notification): OneSignal SDK Integration and Configuration

### LINKED ISSUE #400: OneSignal SDK Integration & Configuration
**Description**
Integrate OneSignal SDK into the Android app. Add OneSignal dependencies to `build.gradle.kts`, configure `AndroidManifest.xml` with required permissions and meta-data, and initialize OneSignal SDK in the app. Verify device registration and token generation through OneSignal dashboard.

### PR DESCRIPTION
**Pull request by Bouazz-a**
## Summary
This PR integrates the OneSignal SDK for push notification delivery. Implements notification processing, device token registration on user authentication, and stores notifications in Firestore for persistence.

### Related issues
Closes #400 and #401, Part of #373

### Key features
- Initialize OneSignal SDK with Android 13+ permission handling
- Process incoming notifications and store in Firestore with type, title, body, and metadata
- Parse 8 notification types (event reminders, invitations, ticket updates, system announcements)
- Auto-register device tokens on sign-in with retry logic (max 3 attempts)
- Auth state listener manages token lifecycle
- Display OnePass logo in notifications

### Tests
**Unit Tests:**
- Notification type parsing for all 8 types and edge cases (null, empty, unknown types)

### CODE REVIEW

File: app/src/main/java/ch/onepass/onepass/service/DeviceTokenManager.kt [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    private val context: Context,
+    private val playerIdProvider: () -> String,
+    private val currentUserIdProvider: () -> String?
+) {
+  private var tokenStoreRetries = 0
+  private val maxRetries = 3
+
+  companion object {
+    private const val TAG = "DeviceTokenManager"
+  }
```
Comment (Zeleph25) [inline app/src/main/java/ch/onepass/onepass/service/DeviceTokenManager.kt]: ## Strengths
- The class implements a proper retry mechanism for the OneSignal player ID, which ensures that token storage is attempted multiple times if the ID is not yet available.
- Null `userId` values are handled safely, preventing crashes when the user is not authenticated.
- Token storage logic is well separated from the UI, making the class easier to maintain and test.
- The use of `suspend` functions for coroutine support is correct and idiomatic.
- Device information and app version are captured properly, ensuring meaningful data is stored in Firestore.

**Review by Zeleph25 — CHANGES_REQUESTED** [review]
### Final Review Summary & Requested Changes

Great work @Bouazz-a !!

The app shows solid practices: clean architecture, good separation of concerns, and well-structured notification handling. OneSignal and Firebase integrations are maintainable.  

**Code coverage is currently 67%** — please share any context on this.  

---

#### Key Recommendations

- Replace hard-coded delays with event-driven mechanisms.  
- Improve coroutine handling to avoid orphaned jobs.  
- Handle privacy-sensitive identifiers carefully (e.g., avoid using non-resettable `ANDROID_ID`).  
- Consider increasing test coverage.  

Following these steps will improve **reliability, maintainability, privacy, and test coverage**.

File: app/src/main/java/ch/onepass/onepass/service/NotificationService.kt [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    /** Processes incoming notification and stores in Firestore. Extracted for testing purposes. */
+    @VisibleForTesting
+    internal fun processNotification(
+        userId: String?,
+        title: String,
+        body: String,
+        additionalData: JSONObject?,
+        repository: NotificationRepository
+    ) {
+      CoroutineScope(Dispatchers.IO).launch {
```
[the remark this answers is not in the document]
Comment (Bouazz-a) [reply app/src/main/java/ch/onepass/onepass/service/NotificationService.kt]: For this specific use case, creating a scope per notification is actually fine, it's just a quick Firestore write that completes in milliseconds. The notification is delivered by OneSignal regardless, and the coroutine just persists it for our in-app center (similar to how Instagram/Twitter handle push notifications).
Since this only runs when users are authenticated and the operation is fire-and-forget, lifecycle management isn't a concern. If we were doing long-running operations or needed cancellation control, I'd agree with using a shared scope.

**Review by Zeleph25 — APPROVED** [review]
Thanks for implementing this @Bouazz-a !! The integration is well done — the code is clean, the structure is easy to follow, and the added documentation is very helpful.

I also want to acknowledge how well you addressed all my previous remarks. For example, the security concern regarding the device ID was fully resolved with your updated approach, and for the points where you provided reasoning instead of changes, your explanations were clear and convincing.

Regarding the 69% line coverage: I initially flagged this as an issue, but after further discussion with the team and the coaches, it’s clear that the lower coverage is mainly due to the `MainActivity` configuration logic, which can’t realistically be tested. Given that context, I’m comfortable approving this.

Overall, great work on this PR — thanks again for the effort and attention to detail. **Approved.**
